apis/apiextensions/v1alpha1/composition_types.go

```diff
@@ -87,6 +87,26 @@ type ComposedTemplate struct {
 	// resource to the composition instance connection secret.
 	// +optional
 	ConnectionDetails []ConnectionDetail `json:"connectionDetails,omitempty"`
+
+	// ReadinessProbe allows users to use a custom readiness check. The default
+	// readiness probe is to have the "Ready" condition to be "True".
+	ReadinessProbe *ReadinessProbe `json:"readinessProbe,omitempty"`
```

Comment:
Naming nit: "probe" to me implies that we'd be "probing" the resource whose readiness we're concerned with - i.e. doing something a little more active like running an exec or making a network request. I'd prefer something like `readinessConstraint`, `readinessRequirement`, or `readinessCheck` here.

Reply:
Also, I think we should support an array of these. I'm very skeptical that inspecting only one field to determine that something is ready will scale over time.